src/cargo/util/errors.rs

```diff
@@ -135,6 +135,47 @@ impl<'a> Iterator for ManifestCauses<'a> {
 
 impl<'a> ::std::iter::FusedIterator for ManifestCauses<'a> {}
 
+/// Error during resolution providing a path of `PackageId`s.
+///
+/// This error adds no displayable info of it's own.
+pub struct ResolveError {
+    cause: Error,
+    package_path: Vec<PackageId>,
+}
+
+impl ResolveError {
```

Comment:
Given that this has now become a resolver only thing, maybe it should go in a `src/cargo/core/resolver/errors.rs`? Then `fn activation_error` can live there asswell. Unless you think that this will still be more widely applicable.

Reply:
Yeah sure, can do.